src/Illuminate/Foundation/Testing/DatabaseTransactions.php

```diff
@@ -33,7 +33,6 @@ public function beginDatabaseTransaction()
                 $connection->unsetEventDispatcher();
                 $connection->rollBack();
                 $connection->setEventDispatcher($dispatcher);
-                $connection->disconnect();
```

Comment:
Fixed in https://github.com/laravel/framework/pull/49327/commits/71d57c294d408fce140e091ca1c71b39f3bb95b8

Immediately after this part is processed by `callBeforeApplicationDestroyedCallbacks`, the newly added processing will also be performed. The intention was to eliminate waste, but since there is no harm in having it, the way you mentioned may have been better in the sense that it does not change the existing behavior. Thanks!